sudoku_board.cc

```diff
@@ -0,0 +1,123 @@
+
+#include "sudoku_board.h"
+
+// ------------------------------- public methods ---------------------------------
+
+SudokuBoard::SudokuBoard(int size){
+    m_size = size;
+    m_board_values = std::vector(size, std::vector<int>(size, EMPTY_CELL_SIGN));
```

Comment:
Use the initialization list. Once you are in the constructor space it means all of the variables have already been initialized, and you don't want that - you want them to be initialized with your values the moment they are created.
For example, add `m_board_value(size, std::vector<int>(size, EMPTY_CELL_SIGN)` to the initialization list.